src/codegen/mod.rs

```diff
@@ -1487,6 +1500,17 @@ impl ToRustTy for Type {
                     }
                 }
             }
+            TypeKind::Complex(fk) => {
```

Comment:
Oh, nevermind, you don't have access to the `result`. Maybe worth moving that to the context inside a `Cell<bool>`, and checking that instead? Even better would be something like `Cell<SeenFlags>`, where `SeenFlags` is a `bitflags!`, but for now moving `saw_union` and `saw_complex` to `BindgenContext` is fine, the other can be a followup if you want.

Otherwise you'll need to check for virtually every possible type, function arguments, etc...